### PR TITLE
Dd/decouple migrations docs

### PR DESCRIPTION
**Pull request by danieldides**
Part of: https://github.com/sourcegraph/sourcegraph/issues/25256

The new database migrations can be run before the service starts up. This PR adds in a quick set up of instructions for admins who may wish to migrate the database before upgrading the application containers.

### CODE REVIEW

File: doc/admin/install/docker-compose/operations.md [hunk 1]
```diff
@@ -186,6 +186,65 @@ docker-compose up -d
...
+
+for DATABASE in frontend codeintel; do
+	docker run --rm --name migrator_$SOURCEGRAPH_VERSION \
+		-e PGHOST='pgsql' \
+		-e PGPORT='5432' \
+		-e PGUSER='sg' \
+		-e PGPASSWORD='sg' \
+		-e PGDATABASE='sg' \
+		-e PGSSLMODE='disable' \
+        -e CODEINTEL_PGHOST='codeintel-db' \
```
Comment (efritz) [inline doc/admin/install/docker-compose/operations.md:226]:
```suggestion
		-e CODEINTEL_PGHOST='codeintel-db' \
```

File: doc/admin/install/docker-compose/operations.md [hunk 2]
```diff
@@ -186,6 +186,65 @@ docker-compose up -d
...
+
+# This will output the current migration version for the codeintel db
+docker exec -it codeintel-db psql -U sg -c "SELECT * FROM codeintel_schema_migrations;"
+##  version   | dirty 
+##------------+-------
+## 1000000030 | f
+##(1 row)
+
+export SOURCEGRAPH_VERSION="122149_2021-12-17_1f7179c"
+
```
Comment (efritz) [inline doc/admin/install/docker-compose/operations.md:216]:
```suggestion
```

File: doc/admin/install/docker-compose/operations.md [hunk 3]
```diff
@@ -186,6 +186,65 @@ docker-compose up -d
...
+### Database Migrations
+
+> NOTE: This feature is only available in versions `3.36` and later
+
+The `migrator` container in the `docker-compose.yaml` file will automatically run on startup and migrate the databases if any changes are required, however administrators may wish to migrate their databases before upgrading the rest of the system when working with large databases. Sourcegraph guarantees database backward compatibility to the most recent minor point release so the database can safely be upgraded before the application code.
+
+To execute the database migrations independently, run the following command (substituting in the version you'd like to migrate to):
+
+```bash
+
```
Comment (efritz) [inline doc/admin/install/docker-compose/operations.md:198]:
```suggestion
```

File: doc/admin/install/docker-compose/operations.md [hunk 4]
```diff
@@ -186,6 +186,65 @@ docker-compose up -d
...
+		-e CODEINTEL_PGPASSWORD='sg' \
+		-e CODEINTEL_PGDATABASE='sg' \
+		-e CODEINTEL_PGSSLMODE='disable' \
+		--network=docker-compose_sourcegraph \
+	sourcegraph/migrator:$SOURCEGRAPH_VERSION \
+	up -db $DATABASE;
+done
+```
+
+> NOTE: These values will work for a standard, docker-compose deployment of Sourcegraph. If you've customized you're deployment (e.g. using an external database service) you will have to modify the environment variables accordingly.
```
Comment (efritz) [inline doc/admin/install/docker-compose/operations.md:238]:
```suggestion
> NOTE: These values will work for a standard, docker-compose deployment of Sourcegraph. If you've customized your deployment (e.g., using an external database service), you will have to modify the environment variables accordingly.
```

File: doc/admin/install/docker-compose/operations.md [hunk 5]
```diff
@@ -186,6 +186,65 @@ docker-compose up -d
 
 You can see what's changed in the [Sourcegraph changelog](../../../CHANGELOG.md).
 
+### Database Migrations
+
+> NOTE: This feature is only available in versions `3.36` and later
+
+The `migrator` container in the `docker-compose.yaml` file will automatically run on startup and migrate the databases if any changes are required, however administrators may wish to migrate their databases before upgrading the rest of the system when working with large databases. Sourcegraph guarantees database backward compatibility to the most recent minor point release so the database can safely be upgraded before the application code.
```
Comment (efritz) [inline doc/admin/install/docker-compose/operations.md:193]: :+1:

**Review by efritz — APPROVED** [review]

**Comment by danieldides** [comment]
Well, I've clearly made a hideous Git mistake trying to rebase this PR...

**Comment by danieldides** [comment]
Closing in favor of: https://github.com/sourcegraph/sourcegraph/pull/29779#pullrequestreview-853474120

**Pull request closed without merging**